astropy/utils/masked/function_helpers.py

```diff
@@ -687,6 +795,14 @@ def quantile(a, q, axis=None, out=None, **kwargs):
     ):
         raise ValueError("Quantiles must be in the range [0, 1]")
 
+    if (interpolation := kwargs.pop("interpolation")) is not None:
+        # we have to duplicate logic from np.quantile here to avoid
+        # passing down the 'interpolation' keyword argument, as it's not
+        # supported by np.lib._function_base_impl._quantile_unchecked
+        kwargs["method"] = np.lib.function_base._check_interpolation_as_method(
```

Comment:
My worry is about numpy >= 2.0 - sorry to not have been clear - where I think it is `np.lib._function_base_impl`.

Reply:
oh, I see, indeed I completely overlooked this while exploring numpy's history, thanks a lot for double checking !